src/main/java/com/conveyal/gtfs/loader/PatternReconciliation.java

```diff
@@ -0,0 +1,455 @@
+            }
+
+            // TODO: write a unit test for this
+            if (firstDifferentIndex == lastDifferentIndex) {
+                throw new IllegalStateException(
+                    "Pattern substitutions are not supported, region of difference must have length > 1.");
+            }
+            String conditionalUpdate;
+
+            // figure out whether a stop was moved left or right
+            // note that if the stop was only moved one position, it's impossible to tell, and also doesn't matter,
+            // because the requisite operations are equivalent
+            int from, to;
+            List<String> newReferenceIds= getPatternReferenceIds(patterns);
+            // Ensure that only a single stop has been moved (i.e. verify stop IDs inside changed region remain unchanged)
+            if (originalPatternIds.get(firstDifferentIndex).equals(patterns.get(lastDifferentIndex).referenceId)) {
+                // Stop was moved from beginning of changed region to end of changed region (-->)
+                from = firstDifferentIndex;
+                to = lastDifferentIndex;
+                verifyInteriorStopsAreUnchanged(originalPatternIds, newReferenceIds, firstDifferentIndex, lastDifferentIndex);
+                conditionalUpdate = String.format("update %s.stop_times set stop_sequence = case " +
+                        // if sequence = fromIndex, update to toIndex.
+                        "when stop_sequence = %d then %d " +
+                        // if sequence is greater than fromIndex and less than or equal to toIndex, decrement
+                        "when stop_sequence > %d AND stop_sequence <= %d then stop_sequence - 1 " +
+                        // Otherwise, sequence remains untouched
+                        "else stop_sequence " +
+                        "end " +
+                        "from %s.trips where %s",
+                    tablePrefix, from, to, from, to, tablePrefix, joinToTrips);
+            } else if (patterns.get(firstDifferentIndex).referenceId.equals(originalPatternIds.get(lastDifferentIndex))) {
+                // Stop was moved from end of changed region to beginning of changed region (<--)
+                from = lastDifferentIndex;
+                to = firstDifferentIndex;
+                verifyInteriorStopsAreUnchanged(originalPatternIds, newReferenceIds, firstDifferentIndex, lastDifferentIndex);
+                conditionalUpdate = String.format("update %s.stop_times set stop_sequence = case " +
+                        // if sequence = fromIndex, update to toIndex.
+                        "when stop_sequence = %d then %d " +
+                        // if sequence is less than fromIndex and greater than or equal to toIndex, increment
+                        "when stop_sequence < %d AND stop_sequence >= %d then stop_sequence + 1 " +
+                        // Otherwise, sequence remains untouched
+                        "else stop_sequence " +
```

Comment:
This, or could we say, as for additions/deletions, `where stop_sequence between <from> and <to> and ...`? (2 instances)

Reply:
Altered to use a single update conditional parameter.